dvc/tree/http.py

```diff
@@ -56,18 +63,47 @@ def __init__(self, repo, config):
         self.ssl_verify = config.get("ssl_verify", True)
         self.method = config.get("method", "POST")
 
+    @memoize
+    def _get_user(self, host):
+        user = ask_username(host)
+        self.path_info.user = user
+        config_level = "local"
+        with self.repo.config.edit(config_level) as conf:
```

Comment:
There have been a few questions about this before, but our remotes don't modify their own config. Some remotes like `gdrive`, save things to .dvc/tmp, but don't modify the config. Looks like we don't really need to do it, right? We can ask again or make the user set `user` in .dvc/config.local by himself. Or am I missing something?

Reply:
I see how it breaks the design a little bit. 
Asking again / asking the user to run a command is a bit unnatural UX (different than git for instance). Can you try to help me find a more appropriate way to save the username as a side effect, that will be satisfying design-wise?

Reply:
@simonlsk Great point. Where does git store the username in such cases?

Reply:
That's a good question, my git installation set my `credential.helper` as `osxkeychain` and I am only prompted once about any combination of `user, host, password`. I thought that you could only fetch a password once you know the `user, host` combination and that the username was being saved in  `.git/config`, but I was wrong. I just read [the code in git](https://github.com/git/git/blob/306ee63a703ad67c54ba1209dc11dd9ea500dc1f/contrib/credential/osxkeychain/git-credential-osxkeychain.c) and it looks like can it fetch entries without knowing the username.
I will try to workaround this, and let you know.